lib/Data/Validate/Sanctions.pm

```diff
@@ -65,25 +63,68 @@ sub is_sanctioned {        ## no critic (RequireArgUnpacking)
 sub _load_data {
     my $self          = shift;
     my $sanction_file = $self->{sanction_file};
-    my $stat          = stat($sanction_file) or croak "Can't get stat of file $sanction_file, please check it.\n";
-    return $self->{_data} if ($stat->mtime <= $self->{last_time} && $self->{_data});
-
-    open(my $fh, '<', $sanction_file) or croak "Can't open file $sanction_file, please check it.\n";
-    my @_data = <$fh>;
-    close($fh);
-    chomp(@_data);
-    $self->{last_time} = $stat->mtime;
-    $self->{_data}     = \@_data;
+    my $_data         = {};
+    my $mtime         = 0;
+
+    if (-e $sanction_file) {
+        $mtime = stat($sanction_file)->mtime or croak "Can't get stat of file $sanction_file, please check it.\n";
+        return $self->{_data} if $mtime <= $self->{last_time} && $self->{_data};
+        open(my $fh, '<', $sanction_file) or croak "Can't open file $sanction_file, please check it.\n";
+        local $/ = undef;
+        $_data = decode_json(<$fh>);
+        close($fh);
+    }
+    $self->{last_time} = $mtime;
+    $self->{_data}     = $_data;
     return $self->{_data};
 }
 
+sub update_data {
+    my $self = shift;
+
+    my $new_data = Data::Validate::Sanctions::Fetcher::run();
+    $self->_load_data();
+    my $updated;
+    foreach my $k (keys %$new_data) {
+        if (ref($self->{_data}{$k}) ne 'HASH' || $self->{_data}{$k}{updated} < $new_data->{$k}{updated}) {
+            $self->{_data}{$k} = $new_data->{$k};
+            $updated = 1;
+        }
+    }
+
+    $self->_save_data if $updated;
+    return;
+}
+
+sub _save_data {
+    my $self = shift;
+
+    my $sanction_file     = $self->{sanction_file};
+    my $new_sanction_file = $sanction_file . ".tmp";
+    my $json              = JSON->new->pretty(1);
+
+    open(my $out_fh, '>:encoding(UTF8)', $new_sanction_file) or die "Can't write to $new_sanction_file: $!\n";
+    print $out_fh $json->encode($self->{_data});
+    close($out_fh);
+
+    rename $new_sanction_file, $sanction_file or die "Can't rename $new_sanction_file to $sanction_file, please check it\n";
+    $self->{last_time} = stat($sanction_file)->mtime;
+    return;
+}
+
 sub _default_sanction_file {
     return $ENV{SANCTION_FILE} if $ENV{SANCTION_FILE};
     my $sanction_file = __FILE__;
-    $sanction_file =~ s/\.pm/\.csv/;
+    $sanction_file =~ s/\.pm/\.json/;
     return $sanction_file;
 }
 
+sub last_updated {
+    my $self = shift;
+    my $list = shift;
```

Comment:
This method can be called without $list, it's defined behaviour:
```
$s->last_updated('OFAC-SDN');
$s->last_updated();
```
both will be handled